node/src/chain.rs

```diff
@@ -427,19 +427,22 @@ pub async fn networks_as_chains(
                 (k, _) => k,
             })
             .expect("each chain should have at least 1 adapter");
-        store
-            .block_store()
-            .chain_store(chain_id)
-            .map(|chain_store| (chain_id, chain_store, adapters, kind))
-            .or_else(|| {
-                error!(
-                    logger,
-                    "No store configured for {} chain {}; ignoring this chain", kind, chain_id
-                );
-                None
-            })
+        (chain_id, adapters, kind)
     });
-    for (chain_id, chain_store, adapters, kind) in chains.into_iter() {
+    for (chain_id, adapters, kind) in chains.into_iter() {
+        let chain_store = match store.chain_store(chain_id) {
+            Some(c) => c,
+            None => {
+                let ident = networks
+                    .chain_identifier(&logger, chain_id)
+                    .await
+                    .expect("must be able to get chain identity to create a store");
+                store
+                    .create_chain_store(chain_id, ident)
+                    .expect("must be able to create store if one is not yet setup for the chain")
```

Comment:
These expects should be errors; otherwise, if you configure a new chain and the provider is down, it won't be possible to start `graph-node` until either the config is changed or the provider comes back. It's fine if the response to the error is to act as if that chain wasn't in the config, but blocking startup entirely in this case is too debilitating.